report_xlsx/report/report_xlsx.py

```diff
@@ -58,4 +69,4 @@ def get_workbook_options(self):
         return {}
 
     def generate_xlsx_report(self, workbook, data, objs):
-        raise NotImplementedError()
+        return
```

Comment:
@apineux why changing this?

Reply:
@sbidoul Because if I keep the raise, the tests will failed. In the test we call create_workbook directly on the ReportXlsxAbstract model instead of create a new model based on ReportXlsxAbstract where we implement the method generate_xlsx_report.

Reply:
@sbidoul @apineux

To mock a method on a model, the best approach is to mock the class from the registry...
```python
report_abstract = self.env["report.report_xlsx.abstract"]
with mock.patch.object(report_abstract.__class__, "generate_xlsx_report") as mocked_method:
    # the test
    ...
    report_abstract. ...

    # and you can check that the method has been called//
    self.assertEqual(mocked_method.call_count, 1)
    
    # if you want you can also check the params received by the method
    mocked_method.assert_called_once_with(.....)
    # or
    mocked_method.assert_called_with(.....)

```

By mocking the method you can let the `NotImplementedError` as default implementation of this abstract method